modules/affinity/src/main/java/org/apache/ignite/internal/affinity/AffinityManager.java

```diff
@@ -85,9 +85,9 @@ public AffinityManager(
         this.baselineMgr = baselineMgr;
         this.vaultMgr = vaultMgr;
 
-        metaStorageMgr.registerWatchByPrefix(new Key(INTERNAL_PREFIX), new WatchListener() {
-            @Override public boolean onUpdate(@NotNull Iterable<WatchEvent> events) {
-                for (WatchEvent evt : events) {
+        metaStorageMgr.registerWatchByPrefix(new ByteArray(INTERNAL_PREFIX), new WatchListener() {
+            @Override public boolean onUpdate(@NotNull WatchEvent watchEvt) {
```

Comment:
What is the correct way to work with WatchEvent? Seems that single() and entryEvent() are never used and we always use for loop.

Reply:
@sanpwc It seems that `entryEvent()` is still useful in case when your watch updates for only one key. So, if we have two different cases, it is possible to have mixed case.